domains/auth/Auth.API/API/Controllers/OidcController.cs

```diff
@@ -115,30 +120,121 @@ private static async Task<UserDescriptor> MapUserDescriptor(IUserDescriptMapper
+        ArgumentException.ThrowIfNullOrEmpty(identityType, nameof(identityType));
+
+        var providerType = GetIdentityProviderEnum(providerName, identityType);
+
+        if (providerOptions.Providers.Contains(providerType) is false)
+        {
+            throw new NotSupportedException("The selected ProviderType is not enabled.");
+        }
+
+        var scope = access.FindFirstValue("scope");
+
+        ArgumentException.ThrowIfNullOrEmpty(scope, nameof(scope));
+
+        string? name = null;
+        string? tin = null;
+        string? companyName = null;
+        var keys = new Dictionary<ProviderKeyType, string>();
+
+        switch (providerType)
+        {
+            case ProviderType.MitID_Professional:
+                // TODO: Make sure this implementation is correct when implementing MitID Erhverv.
+
+                //name = userInfo.FindFirstValue("mitid.identity_name");
+                //tin = userInfo.FindFirstValue("nemlogin.cvr");
+                //companyName = userInfo.FindFirstValue("nemlogin.org_name");
+
+                //keys.Add(ProviderKeyType.RID, userInfo.FindFirstValue("nemlogin.nemid.rid") ?? throw new ArgumentNullException());
+                //keys.Add(ProviderKeyType.EIA, userInfo.FindFirstValue("nemlogin.persistent_professional_id") ?? throw new ArgumentNullException());
+                //keys.Add(ProviderKeyType.MitID_UUID, userInfo.FindFirstValue("mitid.uuid") ?? throw new ArgumentNullException());
+
+                //break;
+
+                throw new NotImplementedException("ProviderType.MitID_Professional hasn't been implemented yet.");
+            case ProviderType.MitID_Private:
+                name = userInfo.FindFirstValue("mitid.identity_name");
+
+                if (userInfo.FindFirstValue("nemid.pid") is not null)
+                {
+                    keys.Add(ProviderKeyType.PID, userInfo.FindFirstValue("nemid.pid")!);
+                }
+                keys.Add(ProviderKeyType.MitID_UUID, userInfo.FindFirstValue("mitid.uuid") ?? throw new ArgumentNullException("mitid.uuid"));
+                break;
+            case ProviderType.NemID_Professional:
+                name = userInfo.FindFirstValue("nemid.common_name");
+                tin = userInfo.FindFirstValue("nemid.cvr");
+                companyName = userInfo.FindFirstValue("nemid.company_name");
+
+                keys.Add(ProviderKeyType.RID, userInfo.FindFirstValue("nemid.ssn") ?? throw new ArgumentNullException("nemid.ssn"));
+                break;
+            case ProviderType.NemID_Private:
+                name = userInfo.FindFirstValue("nemid.common_name");
+
+                keys.Add(ProviderKeyType.MitID_UUID, userInfo.FindFirstValue("nemid.pid") ?? throw new ArgumentNullException("nemid.pid"));
+                break;
+            default:
+                throw new NotSupportedException("An unsupported ProviderType was supplied.");
```

Comment:
Yes, but _which_ provider type should be included in the exception message in `GetIdentityProviderEnum` and see if we can avoid a default case here.